sdk/rust/oak_tests/src/lib.rs

```diff
@@ -22,24 +22,34 @@ use oak_runtime::ChannelEither;
 use protobuf::{Message, ProtobufEnum};
 use std::collections::HashMap;
 use std::process::Command;
-use tempdir::TempDir;
+
+use serde::Deserialize;
 
 // TODO(#544): re-enable unit tests of SDK functionality
 
+#[derive(Deserialize)]
+struct CargoMetadata {
+    target_directory: String,
+}
+
 /// Uses cargo to compile a Rust manifest to Wasm bytes. Compilation is performed in a temporary
 /// directory.
 pub fn compile_rust_wasm(cargo_path: &str, module_name: &str) -> std::io::Result<Vec<u8>> {
-    let temp_dir = TempDir::new("")?;
+    let cargo_output = Command::new("cargo").arg("metadata").output()?;
+    let metadata_json =
+        std::str::from_utf8(&cargo_output.stdout).expect("cargo metadata returned non-utf8");
+    let target_directory = serde_json::from_str::<CargoMetadata>(metadata_json)?.target_directory;
```

Comment:
Would [cargo_metadata](https://docs.rs/cargo_metadata/0.9.1/cargo_metadata/) be clearer and/or easier?

Reply:
Done